pinot-controller/src/main/java/org/apache/pinot/controller/util/TableMetadataReader.java

```diff
@@ -84,11 +85,46 @@ public List<String> getReloadCheckResponses(String tableNameWithType, int timeou
 
   /**
    * This api takes in list of segments for which we need the metadata.
+   * This calls the server to get the metadata for all segments instead of making a call per segment.
    */
   public JsonNode getSegmentsMetadata(String tableNameWithType, List<String> columns, Set<String> segmentsToInclude,
       int timeoutMs)
       throws InvalidConfigException, IOException {
-    return getSegmentsMetadataInternal(tableNameWithType, columns, segmentsToInclude, timeoutMs);
+    return getSegmentsMetadataInternalV2(tableNameWithType, columns, segmentsToInclude, timeoutMs);
+  }
+
+  private JsonNode getSegmentsMetadataInternalV2(String tableNameWithType, List<String> columns,
```

Comment:
Let's modify the `getSegmentsMetadataInternal()` to first try the new API, and fallback to the old way if the new API doesn't work (e.g. server still runs old version)